jetty-http/src/test/java/org/eclipse/jetty/http/QuotedQualityCSVTest.java

```diff
@@ -309,5 +312,44 @@ public void testNoQuality()
         values.addValue("one,two;,three;x=y");
         Assert.assertThat(values.getValues(),Matchers.contains("one","two","three;x=y"));
     }
+
+
+    @Test
+    public void testQuality()
+    {
+        List<String> results = new ArrayList<>();
+
+        QuotedQualityCSV values = new QuotedQualityCSV()
+        {
+            @Override
+            protected void parsedValue(StringBuffer buffer)
+            {
+                super.parsedValue(buffer);
+                results.add("parsedValue: " + buffer.toString());
+            }
+
+            @Override
+            protected void parsedParam(StringBuffer buffer, int valueLength, int paramName, int paramValue)
+            {
+                String param = buffer.substring(paramName, buffer.length());
+                results.add("parsedParam: " + param);
+                super.parsedParam(buffer, valueLength, paramName, paramValue);
+            }
+        };
+
+
+        /*
+        The quality value q=0.5 parameter is treated specially, it is consumed and used as the weight for an empty string value
+        however the parameter p=0.4 is not consumed and is instead added to values
+
+        Note: allowing the q parameter by itself is not specified behaviour in the RFC, but the implementation here is lenient
+        */
+        values.addValue(" en-US;es-ES,q=0.5,p=0.4");
+        assertThat(values,contains("en-US;es-ES", "p=0.4", ""));
+
+        assertThat(results,contains("parsedValue: en-US", "parsedParam: es-ES",
```

Comment:
For the sake of absolute clarity, I'd probably put these one per line with a comment against each saying what it is